### ISSUE TITLE
noImplicitAny+prototype assignment causes incorrect error on prototype methods

### ISSUE THREAD

**Issue by sandersn** [microsoft/TypeScript#26923]
Note that this is only true with noImplicitAny on. The bug does not manifest with it off.
This is a regression in 3.1. It does not manifest in 3.0.

```js
// @noImplicitAny: true
/** @constructor */
var Multimap = function() {
    this._map = {};
};

Multimap.prototype = {
  x() { }
}

Multimap.prototype.lol = function() {
}

var mm = new Multimap();
mm._map
mm.x
mm.lol
```

**Expected behavior:**
No error on `Multimap.prototype.lol`, and `_map`, `x` and `lol` are properties on Multimap.

**Actual behavior:**
Error: 'lol' does not exist on type '{ x(): void }'. But all 3 **are** properties on Multimap.

**Comment by sandersn** [comment]
The solution might be to better understand prototype assignment in getWidenedTypeFromJSPropertyAssignments. I tried calling getJavascriptClassType when there's a prototype assignment, but that resulted in a circularity. Probably a slightly more complex fix will work.

**Comment by sandersn** [comment]
This is now fixed.

**Comment by RyanCavanaugh** [comment]
This doesn't seem like a bug - if there's one initialization with an object literal, we can reasonably assume out-of-domain later assignments are typos.
